lib/Runtime/Library/JavascriptError.cpp

```diff
@@ -159,6 +159,54 @@ namespace Js
     }
 #endif
 
+    Var JavascriptAggregateError::NewAggregateErrorInstance(RecyclableObject* function, CallInfo callInfo, ...)
+    {
+        PROBE_STACK(function->GetScriptContext(), Js::Constants::MinStackDefault);
+        ARGUMENTS(args, callInfo);
+        ScriptContext* scriptContext = function->GetScriptContext();
+        JavascriptLibrary* library = scriptContext->GetLibrary();
+        JavascriptAggregateError* pError = scriptContext->GetLibrary()->CreateAggregateError();
+        Var newTarget = args.GetNewTarget();
+        RecyclableObject* errors = args.Info.Count > 1 ? JavascriptOperators::IterableToList(VarTo<RecyclableObject>(args[1]), scriptContext) : library->CreateArray(0);
+        Var message = args.Info.Count > 2 ? args[2] : library->GetUndefined();
+        pError->SetErrors(errors);
+
+        return JavascriptError::NewInstance(function, pError, callInfo, newTarget, message);
+    }
+
+    Var JavascriptAggregateError::EntryGetterErrors(RecyclableObject* function, CallInfo callInfo, ...)
+    {
+        PROBE_STACK(function->GetScriptContext(), Js::Constants::MinStackDefault);
+
+        ARGUMENTS(args, callInfo);
+        ScriptContext* scriptContext = function->GetScriptContext();
+
+        AssertMsg(args.Info.Count > 0, "Should always have implicit 'this'");
+
+        Assert(!(callInfo.Flags & CallFlags_New));
+
+        if (args[0] == 0 || !JavascriptOperators::IsObject(args[0]))
+        {
+            JavascriptError::ThrowTypeError(scriptContext, JSERR_This_NeedObject, _u("AggregateError.prototype.errors"));
+        }
+
+        if (!VarIs<JavascriptAggregateError>(args[0])) {
+            JavascriptError::ThrowTypeError(scriptContext, JSERR_This_NeedObject, _u("AggregateError.prototype.errors"));
+        }
+
+        JavascriptLibrary* library = scriptContext->GetLibrary();
+        JavascriptAggregateError* thisError = VarTo<JavascriptAggregateError>(args[0]);
+
+        RecyclableObject* iterator = JavascriptOperators::GetIterator(thisError->GetErrors(), scriptContext);
+        JavascriptArray* errorsList = library->CreateArray(0);
+        JavascriptOperators::DoIteratorStepAndValue(iterator, scriptContext, [&](Var next)
+        {
+            JavascriptArray::Push(scriptContext, errorsList, next);
```

Comment:
I have a concern here - per spec the errors slot is a "List" an arbitrary data structure that normal JS can't interact with.

A JS array with its iterator does not preserve this. A user can give you incorrect results if they edit the iterator property of the Array prototype e.g.:

```js
const Agg = new AggregateError([]);
Array.prototype[Symbol.iterator] = function () { throw new Error("BANG"); }
Agg.errors; // throws BANG - but should not do so per spec
```
Probably need to work with something like an SList instead - or perhaps for a smaller change use a for loop and the internal DirectGetItemAt method to access this array without using its iterator.

Reply:
As an extra note - if you go with the DirectGetItemAt method you may also need to create your internal array without a prototype as otherwise a user could perhaps give you odd results by editing the Array.prototype.